apps/sms/views/shared/js/settings.js

```diff
@@ -67,9 +70,37 @@ var Settings = {
       }
     }
 
+    //create lock for SEND_READ_REPORT_KEY
+    keyHandlerSet[this.SEND_READ_REPORT_KEY] = this.toSendReadReport.bind(this);
+
```

Comment:
nit: add it directly in keyHandlerSet above, and call the  function "initSendReadReport" like the other ones